### PR TITLE
chore: added z-user-dropdown menu link click event

### PR DESCRIPTION
**Pull request by pmerolle**
# Chore - Z-User-Dropdown - dropdown menu link click event

----

### Motivation and Context
This PR adds an Event fired on dropdown menu link click to help handle link click (es.: logout)

### Priority

- [ ] 1 - Highest
- [ ] 2 - High
- [x] 3 - Medium
- [ ] 4 - Low
- [ ] 5 - Lowest
- [ ] 6 - Not a Priority

### Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Feature (non-breaking change which adds functionality)
- [ ] Component (add a Component as approved dy Design System)
- [ ] Docs (add documentation)
- [x] Chore (changes that adds small enhancement)
- [ ] Breaking (fix or feature that would cause existing functionality to not work as expected)

### Note

----

## Component and Fix approval flow to Master branch

A Pull Request to be merged must have two approvals, one from a member of the product team who developed it and another from a member of the dst dev team other than the team representative.

### CODE REVIEW

**Review by vpasquino — APPROVED** [review]

File: src/components/navigation/z-user-dropdown/index.tsx [hunk 1]
```diff
@@ -17,7 +17,7 @@ import { mobileBreakpoint } from "../../../constants/breakpoints";
 })
 export class ZUserDropdown {
   /** logged status flag */
-  @Prop() logged?: boolean;
+  @Prop({mutable: true}) logged?: boolean;
```
Comment (vpasquino) [inline src/components/navigation/z-user-dropdown/index.tsx:20]: Why `mutable`?

Comment (pmerolle) [reply src/components/navigation/z-user-dropdown/index.tsx:20]: because we need to update the component when this prop change

Comment (vpasquino) [reply src/components/navigation/z-user-dropdown/index.tsx:20]: > it's possible to explicitly allow a Prop to be mutated from inside the component, by declaring it as mutable
The prop is never changed from inside the component (as if it was a State variable), so I think the declaration is useless.

Comment (pmerolle) [reply src/components/navigation/z-user-dropdown/index.tsx]: ok removed

**Review by vpasquino — APPROVED** [review]

**Review by a-sartori-wt — APPROVED** [review]

**Pull request merged**